### PR TITLE
Fix curve negative energy issue in plant loop heat pumps

### PR DESCRIPTION
**Pull request by jcyuan2020**
Pull request overview
---------------------
 - Warn and reset negative EIR (and capacity) curves to prevent negative energy in the plant loop heat pump module.
 - Stemmed from comments from a previous issue fix [here](https://github.com/NREL/EnergyPlus/pull/9573#issuecomment-1198099296) and the technicality call discussions later on.

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by jcyuan2020** [comment]
I will mark this ready for review now and wait for the ci at the same time. The changes are clear enough so it would expect no diff or just diffs that are easy to explanations.

File: src/EnergyPlus/PlantLoopHeatPumpEIR.cc [hunk 1]
```diff
@@ -244,6 +244,28 @@ void EIRPlantLoopHeatPump::doPhysics(EnergyPlusData &state, Real64 currentLoad)
     // evaluate capacity modifier curve and determine load side heat transfer
     Real64 capacityModifierFuncTemp =
         CurveManager::CurveValue(state, this->capFuncTempCurveIndex, loadSideOutletSetpointTemp, this->sourceSideInletTemp);
+
+    if (capacityModifierFuncTemp < 0.0) {
+        if (this->capModFTErrorIndex == 0) {
+            ShowSevereMessage(state, format("{} \"{}\":", DataPlant::PlantEquipTypeNames[static_cast<int>(this->EIRHPType)], this->name));
```
Comment (Myoldmopar) [inline src/EnergyPlus/PlantLoopHeatPumpEIR.cc:250]: If this pattern gets propagated any further, we should probably refactor it into a worker function that can be reused by many component models.

File: tst/EnergyPlus/unit/PlantLoopHeatPumpEIR.unit.cc [hunk 2]
```diff
@@ -2996,5 +2997,181 @@ TEST_F(EnergyPlusFixture, TestOperatingFlowRates_FullyAutosized_AirSource)
...
+
+    EXPECT_EQ(state->dataErrTracking->TotalWarningErrors, 1);
+    EXPECT_EQ(state->dataErrTracking->TotalSevereErrors, 0);
+    EXPECT_EQ(state->dataErrTracking->LastSevereError, "HeatPump:PlantLoop:EIR:Cooling \"HP COOLING SIDE\":");
+
+    EXPECT_EQ(state->dataErrTracking->RecurringErrors(1).Count, 1);
+    EXPECT_EQ(state->dataErrTracking->RecurringErrors(1).Message,
+              " ** Warning ** HeatPump:PlantLoop:EIR:Cooling \"HP COOLING SIDE\": EIR Modifier curve (function of PLR) output is negative warning "
+              "continues...");
+}
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/PlantLoopHeatPumpEIR.unit.cc:3174]: :+1:

**Review by Myoldmopar — APPROVED** [review]
No issues here, I don't think a local build is necessary, but I'll do it anyway.

**Comment by Myoldmopar** [comment]
CI is all happy except the pixel counting unit test, :+1:

**Comment by Myoldmopar** [comment]
Everything is fine, no reason to hold this.  Thanks @jcyuan2020

**Pull request merged**